File: app-api/LibElectronAppDemo/Database/Models/Track.cs
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace LibElectronAppDemo.Database.Models;

public class Track : NamedEntity
{
    public long? GenreId { get; set; }
    public long ArtistId { get; set; }
    public long AlbumId { get; set; }
    public float Rating { get; set; }
    public DateTimeOffset DateAdded { get; set; }
    public int? DiscNumber { get; set; }
    public int? TrackNumber { get; set; }
    public TimeSpan Duration { get; set; }
    public string Filename { get; set; }
    [NotMapped]
    public Lazy<Genre> Genre { get; set; }
    [NotMapped]
    public Lazy<Artist> Artist { get; set; }
    [NotMapped]
    public Lazy<Album> Album { get; set; }
}

File: app-ui/src/models/demoData/Track.ts
import {NamedEntity} from "./NamedEntity";
import {Genre} from "./Genre";
import {Artist} from "./Artist";
import {Album} from "./Album";
import {LazyShim} from "../LazyShim";

export class Track extends NamedEntity {
    public genreId: number | undefined;
    public artistId: number = 0;
    public albumId: number = 0;
    public rating: number = 0.0;
    public dateAdded: Date = new Date();
    public discNumber: number | undefined;
    public trackNumber: number | undefined;
    public duration: number = 0.0;
    public filename: string = '';
    public genre: LazyShim<Genre> = new LazyShim<Genre>();
    public artist: LazyShim<Artist> = new LazyShim<Artist>();
    public album: LazyShim<Album> = new LazyShim<Album>();
}
